src/main/java/frc/robot/common/hardware/MotorController.java

```diff
@@ -46,14 +47,26 @@ public MotorController(String name, int deviceID, int smartCurrentLimit, boolean
     }
 
     public CANSparkMax getSparkMax() {
+        // Check first that mSparkMax has been instantiated
+        if(mSparkMax == null) {
+            throw new NullPointerException("Spark MAX motor has not been instantiated.");
+        }
         return mSparkMax;
     }
 
     public RelativeEncoder getEncoder() {
+        // Check first that mEncoder has been instantiated
+        if(mEncoder == null) {
+            throw new NullPointerException("Encoder has not been instantiated.");
+        }
         return mEncoder;
     }
 
-    public SparkMaxPIDController getPID() {
+    public SparkMaxPIDController getPID() { 
+        // Check first that mPIDController has been instantiated
+        if(mPIDController == null) {
+            throw new NullPointerException("PID Controller has not been instantiated.");
```

Comment:
Could we add the mName to the error message? Though if mSparkMax doesn't exist, mName probably doesn't either.

Reply:
I think we can get the name of the MotorController, we wouldn't get this far if the MotorController itself didn't exist. Added!